Release: 2023.10.2

syuilo opened the pull request:
### General
- Feat: アンテナでローカルの投稿のみ収集できるようになりました
- Feat: サーバーサイレンス機能が追加されました
- Enhance: 新規にフォローした人の返信をデフォルトでTLに追加できるオプションを追加
- Enhance: HTL/LTL/STLを2023.10.0アップデート以前まで遡れるように
- Enhance: フォロー/フォロー解除したときに過去分のHTLにも含まれる投稿が反映されるように
- Enhance: ローカリゼーションの更新
- Enhance: 依存関係の更新

### Client
- Enhance: TLの返信表示オプションを記憶するように
- Enhance: 投稿されてから時間が経過しているノートであることを視覚的に分かりやすく

### Server
- Enhance: タイムライン取得時のパフォーマンスを向上
- Enhance: ストリーミングAPIのパフォーマンスを向上
- Fix: users/notesでDBから参照した際にチャンネル投稿のみ取得される問題を修正
- Fix: コントロールパネルの設定項目が正しく保存できない問題を修正
- Fix: 管理者権限のロールを持っていても一部のAPIが使用できないことがある問題を修正
- Change: ユーザーのisCatがtrueでも、サーバーではnyaizeが行われなくなりました
  - isCatな場合、クライアントでnyaize処理を行うことを推奨します

syuilo committed:
.js

syuilo committed:
enhance(frontend): TLの返信表示オプションを記憶するように

Resolve #12016

syuilo committed:
perf(backend): nyaizeを投稿時にやる

Resolve #12030

syuilo committed:
perf(backend): tweak populateMyReaction

syuilo committed:
perf(backend): improve streaming api performance (#12033)

* wip

* Update NoteEntityService.ts

* wip

* wip

* wip

* wip

syuilo committed:
Update CHANGELOG.md

syuilo committed:
Update CHANGELOG.md

syuilo committed:
perf(backend): createdAtをidから取得するように & 無駄なDateインスタンスの生成を避けるように

syuilo committed:
fix of 1fa1d31696

syuilo committed:
fix tests

syuilo committed:
fix test

syuilo committed:
feat: local-only antenna

Resolve #11869

syuilo committed:
fix test

mattyatea committed:
feat: サーバーサイレンス機能を追加 (#12031)

* feat : サーバーサイレンスを追加

* Update CHANGELOG.md

* Update CHANGELOG.md

* Update locale

* Update instance-info.vue

* update misskey-js.api.md

* lint fix

* migration fix

* 既存のものを使うように

* fix

* 色々直した

* Update packages/frontend/src/pages/admin/instance-block.vue

* Update packages/frontend/src/pages/admin/instance-block.vue

* Update packages/frontend/src/components/MkInstanceCardMini.vue

* Update packages/backend/src/core/entities/InstanceEntityService.ts

* Update packages/backend/src/core/entities/InstanceEntityService.ts

* Update packages/backend/src/core/entities/InstanceEntityService.ts

* Update packages/backend/src/core/UserFollowingService.ts

* Update packages/backend/src/core/UserFollowingService.ts

* fix: サイレンスされてるサーバーからの投稿は全部ホームにする

* fix: undefinedでfalseを返すようにした

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

syuilo committed:
Update CHANGELOG.md

syuilo committed:
update deps

syuilo committed:
refactor

syuilo committed:
update deps

anatawa12 committed:
新規にフォローした人のwithRepliesをtrueにする機能を追加 (#12048)

* feat: add defaultWithReplies to MiUser

* feat: use defaultWithReplies when creating MiFollowing

* feat: update defaultWithReplies from API

* feat: return defaultWithReplies as a part of $i

* feat(frontend): configure defaultWithReplies

* docs(changelog): 新規にフォローした人のをデフォルトでTL二追加できるように

* fix: typo

* style: fix lint failure

* chore: improve UI text

* chore: make optional params of  UserFollowingService.follow() object

* chore: UserFollowingService.follow() accept withReplies

* chore: add withReplies to MiFollowRequest

* chore: process withReplies for follow request

* feat: accept withReplies on 'following/create' endpoint

* feat: store defaultWithReplies in client store

* Revert "feat: return defaultWithReplies as a part of $i"

This reverts commit f2cc4fe6

* Revert "feat: update defaultWithReplies from API"

This reverts commit 95e3cee6

* Revert "feat: add defaultWithReplies to MiUser"

This reverts commit 9f5ab14d7063532de2b049bc2ed40a15658168f5.

* feat: configuring withReplies in import-following

* feat(frontend): configure withReplies

* fix(frontend): incorrectly showRepliesToOthersInTimeline can be shown

* fix(backend): withReplies of following/create not working

* fix(frontend): importFollowing error

* fix: withReplies is not working with follow import

* fix(frontend): use v-model

* style: fix lint

---------

Co-authored-by: Sayamame-beans <61457993+sayamame-beans@users.noreply.github.com>
Co-authored-by: syuilo <syuilotan@yahoo.co.jp>

syuilo committed:
update deps

kakkokari-gtyih committed:
fix: `admin/update-meta` が保存できない (#12057)

* Fix: invalid param impressumUrl

* Update CHANGELOG.md

woxtu committed:
Remove unnecessary nullish coalescing (#12058)

tai-cha committed:
users/notesでDBから参照した際にチャンネル投稿のみ取得される問題を修正 (#12056)

* fix(backend): ユーザーのノート取得でDBにfallbackした際にチャンネルノートのみになる

* Update CHANGELOG

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

nryeouo committed:
Update ja-JP.yml (#12053)

ポスト

syuilo committed:
Update about-misskey.vue

anatawa12 commented:
投稿時にnyaize、戻さないならchangelogに入れたほうがいいと思いました。(nyaize使ってないユーザにもアンテナ等で影響ありますしその点も含めて)

kanasaki15 commented:
ここでいいのか分かりませんが一応報告させていただきます

サーバーサイレンス機能追加辺りから
コントロールパネル→ユーザー→任意のユーザーを選択すると画像の通りエラーになります

![image](https://github.com/misskey-dev/misskey/assets/19925918/1fb16a2a-f78a-44b7-9abf-bb1bf6f4c001)

syuilo committed:
refactor(backend): rename service

syuilo committed:
enhance(backend): improve fanout tl

Resolve #11958
Resolve #12061

syuilo committed:
Update CHANGELOG.md

syuilo commented:
> ここでいいのか分かりませんが一応報告させていただきます
> 
> サーバーサイレンス機能追加辺りから コントロールパネル→ユーザー→任意のユーザーを選択すると画像の通りエラーになります
> 
> ![image](https://user-images.githubusercontent.com/19925918/276044751-1fb16a2a-f78a-44b7-9abf-bb1bf6f4c001.png)

自分の環境では再現しなかったわね

syuilo commented:
どのようなエラーが返ってきているのか分かると嬉しいかも

kanasaki15 commented:
> どのようなエラーが返ってきているのか分かると嬉しいかも

開発者ツールで覗いたらこんな感じでした
![Uploading IMG_3456.png…]()

syuilo commented:
> > どのようなエラーが返ってきているのか分かると嬉しいかも
> 
> 開発者ツールで覗いたらこんな感じでした ![image](https://user-images.githubusercontent.com/19925918/276077392-7d5394ec-4b30-4381-9585-97178d7fae11.png)

🤯

syuilo committed:
fix(backend): fix admin/get-user-ips error

syuilo commented:
> ここでいいのか分かりませんが一応報告させていただきます
> 
> サーバーサイレンス機能追加辺りから コントロールパネル→ユーザー→任意のユーザーを選択すると画像の通りエラーになります
> 
> ![image](https://user-images.githubusercontent.com/19925918/276044751-1fb16a2a-f78a-44b7-9abf-bb1bf6f4c001.png)

修正しました 🙏🏻 🙏🏻

kanasaki15 commented:
> > ここでいいのか分かりませんが一応報告させていただきます
> > サーバーサイレンス機能追加辺りから コントロールパネル→ユーザー→任意のユーザーを選択すると画像の通りエラーになります
> > ![image](https://user-images.githubusercontent.com/19925918/276044751-1fb16a2a-f78a-44b7-9abf-bb1bf6f4c001.png)
> 
> 修正しました 🙏🏻 🙏🏻

確認しましたありがとうございます！

syuilo committed:
enhance(backend): improve fanout tl for stl

syuilo committed:
fix of 6cc02fee99

syuilo committed:
update test

syuilo committed:
2023.10.2-beta.1

taichanNE30 commented:
> Enhance: 新規にフォローした人のをデフォルトでTLに追加できるように

新規にフォローした人の返信を、かしら

syuilo committed:
New Crowdin updates (#12036)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Thai)

syuilo committed:
Update CHANGELOG.md

syuilo committed:
Merge branch 'develop' of https://github.com/misskey-dev/misskey into develop

taichanNE30 commented:
aiscript widgetのテスト落ちてるけど問題なく追加できるし表示されている内容的にFlakeyな気がする
再走でいいかも

syuilo committed:
perf(backend): reduce needless populateMyReaction calls

syuilo committed:
refactor

syuilo committed:
perf(backend): improve my reaction population performance

syuilo committed:
perf(backend): ノートのリアクション情報をキャッシュすることでDBへのクエリを削減

syuilo committed:
fix(backend): fix of 1671575d5d

syuilo committed:
chore: disable debug log of fastify

syuilo committed:
refactor and perf tweak

syuilo committed:
enhance: nyaizeはクライアントで表示時に行うように

Resolve #12030

syuilo committed:
Update CHANGELOG.md

syuilo committed:
2023.10.2-beta.2

syuilo commented:
cypress落ちすぎ

syuilo committed:
fix(backend): Redisがからのときにhybrid-timelineにwithReplies = trueでアクセスするとSQLのシンタックスエラーになる

Fix #12064

syuilo commented:
特に問題なさそうかしら

anatawa12 committed:
fix: replies are included even if withReplies = false in local timeline (#12074)

syuilo commented:
ん、cypress落ちるのは確定ぽいわね
https://github.com/misskey-dev/misskey/pull/12060/commits/30efd932a5828def2cd394e65d333fdbdb447231 から落ちるようになったっぽいけど謎

syuilo commented:
全然関係ないように思えるけど

syuilo commented:
a-

syuilo committed:
fix(frontend): fix of 30efd932a5

syuilo commented:
多分直った

na2na-p committed:
UserLite.nameにnullが入りうるのを型で明示 (#12073)

* UserLite.nameにnullが入りうるのを型で明示

* ドキュメントの追従

anatawa12 committed:
chore: STLのdb fallbackでwithRepliesがtrueのときにすべてのリプライを除外しないように (#12075)

MiFollowingを見るのは実装コストが高いため現状実装していない

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

dependabot[bot] committed:
chore(deps): bump actions/checkout from 4.1.0 to 4.1.1 (#12062)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.0 to 4.1.1.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4.1.0...v4.1.1)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

atsu1125 committed:
fix(frontend): Recieve Unrenote on streaming (#12079)

* fix(frontend): Recieve Unrenote

表示しているリノートがリノート解除されたらストリーミングで受信してすぐに消えるようにする

* fix(frontend): Recieve Unrenote lint fixing

* fix(frontend): Recieve Unrenote Decapture

Decapture忘れてたー

syuilo commented:
@misskey-dev/dev ご確認お願いします

anatawa12 commented:
部外者が失礼します STLも遡れるようにしたはずなので Changelog更新した方がいいと思いました

syuilo committed:
Update CHANGELOG.md

syuilo committed:
New Crowdin updates (#12070)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Traditional)

syuilo committed:
update deps

syuilo committed:
disable cypress widgets tests

syuilo committed:
fix(backend): 管理者権限のロールを持っていても一部のAPIが使用できないことがある問題を修正

taichanNE30 commented on packages/backend/src/core/NoteCreateService.ts:
```diff
@@ -54,7 +54,9 @@ import { RoleService } from '@/core/RoleService.js';
 import { FeaturedService } from '@/core/FeaturedService.js';
-import { RedisTimelineService } from '@/core/RedisTimelineService.js';
+import { FunoutTimelineService } from '@/core/FunoutTimelineService.js';
+import { nyaize } from '@/misc/nyaize.js';
```
unused

taichanNE30 commented:
みています

taichanNE30 commented on packages/backend/src/core/entities/InstanceEntityService.ts:
```diff
@@ -3,9 +3,8 @@
  */
 
-import { Injectable } from '@nestjs/common';
+import { Inject, Injectable } from '@nestjs/common';
```
unused

taichanNE30 commented on packages/backend/src/server/api/endpoints/notes/user-list-timeline.ts:
Brackets, QueryService, redisForTimelinesは不使用だと思います

tai-cha reviewed (commented):
細かいですがよろしくお願いします:pray:

syuilo committed:
clean up

syuilo committed:
fix(backend): 自分のフォロワー限定ノートがWebsoketに乗ってこない

Fix #12081

syuilo committed:
add timeline tests

syuilo committed:
add timeline tests

fruitriin commented:
CHANGE LOGにiOS15だと正常に動かないので16以上が必須と書き加えてもらえるとうれしい

syuilo commented:
それは昔からだから要らなそう

syuilo committed:
chore: tweak MkNotifications

syuilo committed:
enhance(frontend): 投稿されてから時間が経過しているノートであることを視覚的に分かりやすく

syuilo committed:
fix(backend): フォロ限へのリプのリプ元が（非公開）の表示のまま

Fix #12083

syuilo committed:
update test

taichanNE30 commented on packages/backend/test/e2e/streaming.ts:
```diff
@@ -125,6 +135,30 @@ describe('Streaming', () => {
+				const fired = await waitFire(
+					ayano, 'homeTimeline',		// ayano:home
+					() => api('notes/create', { text: 'bar', visibility: 'followers', replyId: note.body.id }, kyoko),	// kyoko posts
+					msg => msg.type === 'note' && msg.body.userId === kyoko.id && msg.body.reply.text === 'foo',
```
msg.body.userIdはayano.idが正しくてmsg.body.reply.userIdはkyoko.idが正しいからだったりしません？これ

syuilo replied:
そんなことはないはず

taichanNE30 replied:
普通に読み間違えてました:pray:

syuilo committed:
update deps

tai-cha committed:
chore: Pull Request時にapi.jsonの差分を表示するActions workflow (#12090)

* chore: Pull Request時にapi.jsonのdiffを出力するworkflow

* refactor: job names

* fix: set repository to get api diff

* chore: set permission to workflow

* set sleep 30s (shorter)

* chore: set label of diff

* chore: more attempts to fetch misskey

* chore: add full diff output of api.js

* chore: save full-diff to Artifact

* chore: add message to download diff Artifact

shiosyakeyakini-info committed:
fix(backend) api/iの未読のお知らせにcreatedAtを追加するように修正 (#12092)

Co-authored-by: sorairo <sorairo@shiosyakeyakini.info>

siketyan committed:
fix(frontend): Use opening quote in notifications (#12082)

Signed-off-by: Natsuki Ikeguchi <me@s6n.jp>

syuilo commented:
マージするか

taichanNE30 commented:
（マージ前にbeta外すの一応言っておきます…まだbeta）

woxtu committed:
Switch avatar images that depend on the animation setting (#12097)

syuilo committed:
2023.10.2

syuilo committed:
New Crowdin updates (#12094)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Uyghur)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Croatian)

* New translations ja-jp.yml (Haitian Creole)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Simplified)

syuilo commented:
@acid-chicken マージしちゃう

zyoshoka commented on CHANGELOG.md:
```diff
@@ -12,6 +12,30 @@
 
 -->
 
+## 2023.x.x (unreleased)
```
書き換え忘れていませんか？

syuilo replied:
🤯

syuilo commented:
🙏🏻 🙏🏻 🙏🏻
